### PR TITLE
TDL-9801: `payouts` incorrectly mapped to `transfers` object results in transfer objects replicated by payouts stream

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-9801: `payouts` incorrectly mapped to `transfers` object results in transfer objects replicated by payouts stream
- Added **payout** with **transfer** as object type for payout events, to replicate payout events with newer and older API versions.

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/unittests/test_payout_events_object.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        # function call
+        should_sync = should_sync_event(MockPayout({"id": "po_test123", "object": "transfer"}), STREAM_TO_TYPE_FILTER.get("payouts").get("object"), {})
+        # verify if the Tap will sync the payout events
+        self.assertTrue(should_sync)
+
+    def test_payout_stream_payout_object(self):
+        """
+            Test cases to verify the Tap syncs payout events with 'object' type 'payout'
+        """
+        # funcution call
```
Comment (savan-chovatiya) [inline tests/unittests/test_payout_events_object.py:39]:
```suggestion
        # function call
```

**Review by savan-chovatiya — CHANGES_REQUESTED** [review]

File: tests/test_all_fields.py [hunk 2]
```diff
@@ -548,8 +572,22 @@ def all_fields_test(self, streams_to_test):
                                         f"AssertionError({failure_1})")
 
                                     nested_key = KNOWN_NESTED_MISSING_FIELDS.get(stream, {})
-                                    if self.find_nested_key(nested_key, expected_field_value, field):
-                                        continue
+                                    # Check whether expected_field_value is list or not.
+                                    # If expected_field_value is list then loop through each item of list
+                                    if type(expected_field_value) == list:
```
Comment (dbshah1212) [inline tests/test_all_fields.py:577]: Make a separate function for this and add some comments.

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

File: tests/test_all_fields.py [hunk 3]
```diff
@@ -548,8 +585,14 @@ def all_fields_test(self, streams_to_test):
...
+                                        if self.handle_list_data(expected_field_value, field, nested_key):
+                                            continue
+                                    else:
+                                        if self.find_nested_key(nested_key, expected_field_value, field):
+                                            continue
```
Comment (kspeer825) [inline tests/test_all_fields.py:595]: I do not understand what `handle_list_data` and `find_nested_key` are trying to accomplish. All I see them do is skip the checks below. I think it adds complexity to the test and removes the test coverage below and the logged known failures.

**Review by kspeer825 — CHANGES_REQUESTED** [review]

File: tests/test_all_fields.py (hunk 3 above)
Comment (harshpatel4crest) [reply tests/test_all_fields.py:595]: @kspeer There were 2 types of nested fields replicated from the API, 1. dict and 2. list. So, here we have handled dict keys in `find_nested_key` and ignore the failing nested fields and for list fields, we loop over the fields and ignore failing nested fields. Previously we had skipped the first-level field directly so we added the above change to ignore only failed nested fields.

Comment (kspeer825) [reply tests/test_all_fields.py:595]: We were already doing that based on hardcoding expectations. Now if new fields start failing we will not know about it. We should purposefully document and skip discrepant fields if there subfields differ using the already implemented hardcoding approach. We should do this programmatically.

**Review by KrisPersonal — APPROVED** [review]

**Review by RushT007 — CHANGES_REQUESTED** [review]
CCi jobs failing, please fix the issues.

**Comment by harshpatel4crest** [comment]
> CCi jobs failing, please fix the issues.

Fixed the issue. The build is passing now.

File: tests/test_all_fields.py (hunk 3 above)
Comment (harshpatel4crest) [reply tests/test_all_fields.py]: Got it @kspeer825. Update the all fields test case.

**Review by RushT007 — APPROVED** [review]

**Pull request merged**